src/rinterface_extra.c

```diff
@@ -73,8 +73,8 @@ enum igraph_versions {
 void R_check_int_scalar(SEXP value)
 {
   if (Rf_xlength(value) != 1) {
-    igraph_errorf("Expecting a scalar integer but received a vector of length %zu.",
-                    __FILE__, __LINE__, IGRAPH_EINVAL, (size_t) Rf_xlength(value));
+    igraph_errorf("Expecting a scalar integer but received a vector of length %d.",
+                    __FILE__, __LINE__, IGRAPH_EINVAL, Rf_xlength(value));
```

Comment:
Why did you make this change? `%d` is only for `int`, and the return type of `Rf_xlength()` is system-dependent (may not be compatible with `int`!), but _always fits in `size_t`_. I chose `%zu` and a cast to `size_t` in purpose.

Reply:
`zu` is not compatible with some versions of compiler. Maybe good solution will be to use `lu`?

Reply:
What specific compiler is it not compatible with?

Reply:
`zu` is standard in both C99 and C++11. Could it be that the proper standard is not set? I'd be very surprised to see any still-relevant compiler not support it.

Reply:
Works with Clang 3.4 and GCC 4.8. igraph itself wouldn't support anything earlier.

Reply:
> According to tests wit Godbolt, works with Clang 3.4 and GCC 4.8. igraph itself wouldn't support anything earlier.

Here: https://github.com/igraph/rigraph/actions/runs/7512815244/job/20453943297#step:9:84
Maybe for windows there is really wrong flags. I will check.

Reply:
Can you check what was the precise compilation command? Was there an `-std` flag?

Reply:
I see you are casting to `long int`, but `%lu` is actually `unsigned long`.

But please do check what the compilation command is. `%zu` has been part of the standard for a very long time. There is no way that the compiler does not support it. More likely, the wrong standard was chosen.

Can you try separately with the same GCC version? Does it accept `%zu`?

Reply:
> I see you are casting to `long int`, but `%lu` is actually `unsigned long`.
> 
> But please do check what the compilation command is. `%zu` has been part of the standard for a very long time. There is no way that the compiler does not support it. More likely, the wrong standard was chosen.
> 
> Can you try separately with the same GCC version? Does it accept `%zu`?

Oh, yes. Regarding `int` it's typo.

I checked the compile flags. And also added `-std=c99`, but I didn't help. Still see the warning:
```gcc  -I"C:/PROGRA~1/R/R-43~1.1/include" -DNDEBUG -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include/libxml2" -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include" -DLIBXML_STATIC -DUSING_R -I. -Ivendor -Ivendor/cigraph/src -Ivendor/cigraph/include -Ivendor/cigraph/vendor -DNDEBUG -DNTIMER -DNPRINT -DINTERNAL_ARPACK -DIGRAPH_THREAD_LOCAL= -DPRPACK_IGRAPH_SUPPORT -D_GNU_SOURCE=1 -std=c99  -I"C:/rtools43/x86_64-w64-mingw32.static.posix/include"     -O2 -Wall  -mfpmath=sse -msse2 -mstackrealign  -c rinterface_extra.c -o rinterface_extra.o```

using C compiler: 'gcc.exe (GCC) 12.3.0'.

Reply:
@szhorvat https://stackoverflow.com/questions/68900199/how-to-get-mingw-gcc-to-recognize-the-zu-format-specifier-for-size-t

Hm, looks like there still no agreement for `zu`.

Reply:
What that SO answer is saying is that it does actually work but MinGW's GCC _incorrectly_ warns about it. That is very disappointing.

You should be aware that on Windows the `long` type is 32-bit long. Using `long` here is actually wrong. Can't we just use `-Wno-format` specifically on Windows?

Reply:
@Antonov548 Can you try to use `PRIuPTR` instead? Here is an example:

https://godbolt.org/z/eMjM3nz9b

Reply:
Seems `uintptr_t` works well. Let's see in ci. 
I guess it's better to leave all warning enabled.

Reply:
Yes, it's better to leave warnings enabled on all platforms.

Someone just told me about the `__USE_MINGW_ANSI_STDIO` macro. Defining this to either `1` or `0` may help (unclear).

But if `PRIuPTR` works, let's just stick to that, as it is standard.